core/server/data/migrations/versions/2.27/3-add-subdirectory-to-relative-canonical-urls.js

```diff
@@ -0,0 +1,104 @@
+const models = require('../../../../models');
+const common = require('../../../../lib/common');
+const config = require('../../../../config');
+const {URL} = require('url');
+
+module.exports.config = {
+    transaction: true
+};
+
+// We've been incorrectly saving canonical_url fields without a subdirectory.
+// We need this column to be consistent with all other relative URLs, so...
+
+// if we have a configured url with a subdirectory
+// find all posts that have a canonical_url starting with / but not //
+// prefix the subdirectory to the canonical_url and save the post
+module.exports.up = (options) => {
+    // normalize config url to always have a trailing-slash
+    let configUrl = config.get('url');
+    if (!configUrl.endsWith('/')) {
+        configUrl = `${configUrl}/`;
+    }
+    const url = new URL(config.get('url'));
```

Comment:
```suggestion
    const url = new URL(configUrl);
```

Reply:
its updated in `down` transaction but got missed here 😄

Reply:
Good catch, that'll teach me for trying to push a fix too quickly 😄